components/partials/common/derivatives/position.vue

```diff
@@ -559,7 +565,8 @@ export default Vue.extend({
               ? DerivativeOrderSide.Sell
               : DerivativeOrderSide.Buy,
           price: actualPrice,
-          quantity: new BigNumberInBase(position.quantity)
+          quantity: new BigNumberInBase(position.quantity),
+          reduceOnlyOrders: reduceOnlyCurrentOrders
```

Comment:
Decouple this into two separate methods, one when we have `reduceOnlyOrders` and one when we don't. The `handleClosePosition` can call one of these functions (we use similar approach  when we submit limit/market orders in the `trade.vue` component)

Reply:
decoupled the two methods into `closePosition` and `closePositionAndReduceOnly`. `handleClosePosition` uses an early return for `closePositionAndReduceOnly` conditionally if `hasReduceOnlyOrders` computed property is true.  Appreciate the example in `trade.vue`.